packages/client/src/comp/AddressInputs.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { RefObject } from "react"
+
+import { BaseInput } from "./util/Input"
+import styled from "styled-components"
+import { cssQuery } from "./util/cssQuery"
+import { InputId, InputData } from "./states/BaseContainer"
+import { Input } from "muicss/react"
+import { AddressInputId } from "./states/AddressFormContainer"
+
+const FlexBox = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  flex-wrap: wrap;
+  align-items: center;
+  margin: 1em -0.5em;
+`
+
+const FlexGrow = styled.div`
+  margin: 0 0.5em;
+`
+
+const FlexGrowFull = styled.div`
+  margin: 0 0.5em;
+  flex: 0.5;
+  ${cssQuery.small} {
+    flex: 1;
+  }
+`
+
+const FlexGrowHalf = styled.div`
+  margin: 0 0.5em;
+  flex: 0.25;
+  ${cssQuery.small} {
+    flex: 0.5;
+  }
+`
+
+interface Props {
+  streetAddressRef?: RefObject<Input>
+  defaultState?: string
+  defaultZip?: string
+  fields: Record<InputId, InputData> | Record<AddressInputId, InputData>
+  updateField: (id: InputId | AddressInputId, value: string) => void
+}
+
+export const AddressInputs: React.FC<Props> = ({
+  streetAddressRef,
+  defaultState,
+  defaultZip,
+  fields,
+  updateField
+}) => {
+  return (
+    <>
+      <FlexBox id='mailing'>
+        <FlexGrow style={{ flex: 0.75 }}>
+          <BaseInput
+            id="street-address" // This id is used for Warning Box to fill form quickly
+            label="Street address"
+            ref={streetAddressRef || null}
+            required
```

Comment:
I think whether or not it's required depends (for the mailing address) on whether or not the box was checked.  Can you double-check that you can still submit a form without specifying the mailing address?

Reply:
yes you can